executor/tests/test_executor.py

```diff
@@ -0,0 +1,417 @@
+        task_id = get_random_task_id()
+        task = {'task_id': {'value': task_id}}
+
+        stdout_name = 'build/stdout.' + str(task_id)
+        stderr_name = 'build/stderr.' + str(task_id)
+
+        stdout = open(stdout_name, 'w+')
+        stderr = open(stderr_name, 'w+')
+
+        try:
+            command = 'sleep 100'
+            process = subprocess.Popen(command, shell=True, stdout=stdout, stderr=stderr)
+            process_info = process, stdout, stderr
+            shutdown_grace_period_ms = 2000
+
+            stop_signal = Event()
+            stop_signal.set()
+
+            result = ce.await_process_completion(driver, task, stop_signal, process_info, shutdown_grace_period_ms)
+
+            self.assertTrue(result)
+
+            self.assertEqual(1, driver.method_call_count())
+            self.assertEqual(1, len(driver.statuses))
+            actual_status = driver.statuses[0]
+            expected_status = {'task_id': {'value': task_id},
+                               'state': cook.TASK_KILLED}
+            assert_status(expected_status, actual_status)
+
+            self.assertTrue(stdout.closed)
+            self.assertTrue(stderr.closed)
+
+        finally:
+            cleanup_output(stdout_name, stderr_name)
+
+    def manage_task_runner(self, command, assertions_fn):
+        driver = FakeMesosExecutorDriver()
+        task_id = get_random_task_id()
+        task = {'task_id': {'value': task_id},
+                'data': encode_data(json.dumps({'command': command}).encode('utf8'))}
+
+        stdout_name = 'build/stdout.' + str(task_id)
+        stderr_name = 'build/stderr.' + str(task_id)
+
+        stop_signal = Event()
+        completed_signal = Event()
+        max_message_length = 300
+        progress_sample_interval_ms = 100
+        sandbox_location = '/location/to/task/sandbox/{}'.format(task_id)
+        progress_output_name = stdout_name
+        progress_regex_string = '\^\^\^\^JOB-PROGRESS: (\d*)(?: )?(.*)'
+        config = cc.ExecutorConfig(max_message_length=max_message_length,
+                                   progress_output_name=progress_output_name,
+                                   progress_regex_string=progress_regex_string,
+                                   progress_sample_interval_ms=progress_sample_interval_ms,
+                                   sandbox_location=sandbox_location)
+
+        try:
+
+            ce.manage_task(driver, task, stop_signal, completed_signal, config, stdout_name, stderr_name)
+
+            self.assertTrue(completed_signal.isSet())
+            assertions_fn(driver, task_id, sandbox_location)
+
+        finally:
+            cleanup_output(stdout_name, stderr_name)
+
+    def test_manage_task_successful_exit(self):
+        def assertions(driver, task_id, sandbox_location):
+            logging.info('driver.method_call_count() = {}'.format(driver.method_call_count()))
+            self.assertEqual(4, driver.method_call_count())
```

Comment:
This count is the bare minimum messages/statuses sent from the executor: exit-code, sandbox location, running and finished. Didn't need to run it to get this value.